File: 1512-number-of-good-pairs/1512-number-of-good-pairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        HashMap<Integer, Integer> count = new HashMap<>();
        int result = 0;

        for (int num : nums) {
            count.put(num, count.getOrDefault(num, 0) + 1);
            result += count.get(num) - 1;
        }

        return result;
    }
}

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function(nums) {
    let goodPair = 0;
    let map = {};

    for(let i=0;i<nums.length; i++) {
        let item = nums[i];
        if(map[item]) {
            goodPair +=map[item];
            map[item] = map[item]+1;
        } else {
            map[item] = 1;
        }
    }
    return goodPair;
};